src/frame/staking.rs

```diff
@@ -0,0 +1,313 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of substrate-subxt.
+//
+// subxt is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// subxt is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with substrate-subxt.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Implements support for the frame_staking module.
+
+use super::system::{
+    System,
+    SystemEventsDecoder as _,
+};
+use codec::{
+    Decode,
+    Encode,
+    HasCompact,
+};
+use sp_runtime::Perbill;
+use std::{
+    fmt::Debug,
+    marker::PhantomData,
+};
+
+/// A record of the nominations made by a specific account.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, Debug, Ord, PartialOrd, Hash)]
+pub struct Nominations<AccountId> {
+    /// The targets of nomination.
+    pub targets: Vec<AccountId>,
+    /// The era the nominations were submitted.
+    ///
+    /// Except for initial nominations which are considered submitted at era 0.
+    pub submitted_in: EraIndex,
+    /// Whether the nominations have been suppressed.
+    pub suppressed: bool,
+}
+
+/// Information regarding the active era (era in used in session).
+#[derive(Encode, Decode, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct ActiveEraInfo {
+    /// Index of era.
+    pub index: EraIndex,
+    /// Moment of start expresed as millisecond from `$UNIX_EPOCH`.
+    ///
+    /// Start can be none if start hasn't been set for the era yet,
+    /// Start is set on the first on_finalize of the era to guarantee usage of `Time`.
+    pub start: Option<u64>,
+}
+
+/// Data type used to index nominators in the compact type
+pub type NominatorIndex = u32;
```

Comment:
Why is this and ValidatorIndex concrete types, whereas Session::SessionIndex is an associated type?

Reply:
Fixed